cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredTable.java

```diff
@@ -104,6 +109,46 @@ public void upsert(Collection<Field<?>> fields) throws InvalidFieldException {
     return new LimitIterator(new ScannerIterator(scanner, schema), limit);
   }
 
+  @Override
+  public boolean compareAndSwap(Collection<Field<?>> keys, Field<?> oldValue, Field<?> newValue) {
+    LOG.trace("Table {}: CompareAndSwap with keys {}, oldValue {}, newValue {}", schema.getTableId(), keys,
+              oldValue, newValue);
+    if (oldValue.getFieldType() != newValue.getFieldType()) {
+      throw new IllegalArgumentException("Field types of oldValue and newValue are not the same");
+    }
+    if (schema.getType(oldValue.getName()) == null) {
+      throw new InvalidFieldException(schema.getTableId(), oldValue.getName());
+    }
+    if (oldValue.getFieldType() != newValue.getFieldType()) {
```

Comment:
I think this should compare to schema..getType(oldValue)

Reply:
Good catch